spec/operators/elementat-spec.js

```diff
@@ -0,0 +1,47 @@
+/* globals describe, it, expect, hot, cold, expectObservable */
+var Rx = require('../../dist/cjs/Rx');
+var Observable = Rx.Observable;
+
+describe('Observable.prototype.elementAt', function() {
+  it("should return first element by zero-based index", function() {
+    var source = hot('--a--b--c--|', { a: 'a', b: 'b', c: 'c' });
+    var expected =   '--x--|';
+		
+    expectObservable(source.elementAt(0)).toBe(expected, { x: 'a' });
+  });
+	
+  it("should return non-first element by zero-based index", function() {
+    var source = hot('--a--b--c--d--e--f--|', { a: 'a', b: 'b', c: 'c', d: 'd', e: 'e', f: 'f' });
+    var expected =   '-----------x--|';
+		
+    expectObservable(source.elementAt(3)).toBe(expected, { x: 'd' });
+  });
+  
+  it("should return last element by zero-based index", function() {
+    var source = hot('--a--b--c--|', { a: 'a', b: 'b', c: 'c' });
+    var expected =   '--------x--|';
+		
+    expectObservable(source.elementAt(2)).toBe(expected, { x: 'c' });
+  });
+  
+  it("should throw if index is smaller than zero", function() {
+    expect(function() { Observable.range(0,10).elementAt(-1); })
+      .toThrow(new Error('Argument out of range'));
+  });
+	
+  it("should raise error if index is out of range but does not have default value", function() {
+    var source = hot('--a--|', { a: 'a' });
+    var expected =   '-----#';
+    
+    expectObservable(source.elementAt(3))
+      .toBe(expected, null, 'Argument out of range');
+  });
+	
+  it("should return default value if index is out of range", function() {
+    var source = hot('--a--|', { a: 'a' });
+    var expected =   '-----x|';
+    var defaultValue = '42';
+    
+    //expectObservable(source.elementAt(3, defaultValue)).toBe(expected, { x: defaultValue });
```

Comment:
Tried to follow test case description using marble diagrams but having some trouble with specific case. In case of this, actual result is `42|complete()` arrives in same frame. Seems this is related with issue https://github.com/ReactiveX/RxJS/issues/304 - am I understand correctly?